### PR TITLE
[Issue 534] Set up HTTPS for beta

### LINKED ISSUE #534: [Task]: Load cert to load balancer
### Summary

In order to configure TLS for https on beta.grants.gov we need to use aws ACM to load the TLS secret and cert onto the dev load balancer.
This should be configured in terraform to load the data from #533 
However, if we are running out of time before the launch, this can be loaded manually in the console and added to terraform later.

Blocked by #533 

### Acceptance criteria

Dev Load balancer has tls secret and certificate

### PR DESCRIPTION
**Pull request by SammySteiner**
## Summary
Fixes #534 

## Changes proposed
- Load cert to load balancer for frontend by domain
- Create HTTPs listener and forward/redirect rules
- Security group ingress for port 443

## Context for reviewers
- We will need to apply account changes first to get acm imported via the networks/main.tf module.
- We need to make sure the backend won't break without a cert
- We need to make sure to redirect via https if the cert exists

### CODE REVIEW

**Comment by daphnegold** [comment]
I am going to apply these changes, but I think we will have issues dealing with frontend dev and prod with these changes, and need some logic around that

**Comment by daphnegold** [comment]
I am going to apply these changes, but I think we will have issues dealing with frontend dev and prod with this implementation and need some logic around that

**Comment by daphnegold** [comment]
We will have some issues with frontend dev & prod with these changes, we should maybe add some logic around that before merging.

**Comment by daphnegold** [comment]
We added redirect logic, but there's a platform ticket and comment to do this. Maybe we should consider that instead.
```
  # TODO(https://github.com/navapbc/template-infra/issues/163) Disallow incoming traffic to port 80
  # checkov:skip=CKV_AWS_260:Disallow ingress from 0.0.0.0:0 to port 80 when implementing HTTPS support in issue #163
```

**Comment by daphnegold** [comment]
Not all environments have SSL configured, so we only want to allow ingress on 443 if it does. We want to use security group rules to conditionally apply ingress to 443. Because Terraform permits us to create security group ingress directly in `aws_security_group` and `aws_security_group_rule`, there's a kind of surprising issue where using a mix of these approaches to allow for ingress on HTTPS as a rule but HTTP in the security group itself would cause `terraform apply` to flip flop adding and removing ingress on HTTPS.

Solution: Only use security group rules -- move HTTP config from security group into rule. This led to an issue with `terraform apply` spitting out a duplicate security group error and could not proceed. Solving this required modifying Terraform state. 

Steps taken from frontend service directory:
```
terraform state pull > backup.tfstate # ensure that you have a backup of tfstate if things go badly
terraform state list | grep aws_security_group # find all state associated with security group
terraform state rm module.service.aws_security_group.alb
make infra-update-app-service APP_NAME=frontend ENVIRONMENT=dev
```

This creates a new alb security group in AWS, the old security group will still remain and not be referenced by Terraform any longer. Deleted from the console.
Screenshot 2023-09-26 at 11 22 37 AM
Screenshot 2023-09-26 at 11 22 25 AM

**Review by daphnegold — APPROVED** [review]

File: infra/frontend/service/main.tf [hunk 1]
```diff
@@ -89,6 +89,13 @@ data "aws_ssm_parameter" "incident_management_service_integration_url" {
   name  = local.incident_management_service_integration_config.integration_url_param_name
 }
 
+data "aws_acm_certificate" "frontend_cert" {
+  domain = "beta.grants.gov"
```
Comment (acouch) [inline infra/frontend/service/main.tf:93]: This should be updated before merging.

**Pull request merged**